cli/cmd/component-delete.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/pkg/action"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+	"github.com/kinvolk/lokomotive/pkg/config"
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
+)
+
+var componentDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete an installed component",
+	Long: `Delete a component.
+When run with no arguments, all components listed in the configuration are deleted.`,
+	Run: runDelete,
+}
+
+var deleteNamespace bool
+
+// nolint:gochecknoinits
+func init() {
+	componentCmd.AddCommand(componentDeleteCmd)
+	pf := componentDeleteCmd.PersistentFlags()
+	pf.BoolVarP(&deleteNamespace, "delete-namespace", "", false, "Delete namespace with component.")
+}
+
+func runDelete(cmd *cobra.Command, args []string) {
+	contextLogger := log.WithFields(log.Fields{
+		"command": "lokoctl component delete",
+		"args":    args,
+	})
+
+	lokoCfg, diags := getLokoConfig()
+	if len(diags) > 0 {
+		contextLogger.Fatal(diags)
+	}
+
+	componentsToDelete := make([]string, len(args))
+	copy(componentsToDelete, args)
+
+	if len(args) == 0 {
+		componentsToDelete = make([]string, len(lokoCfg.RootConfig.Components))
+
+		for i, component := range lokoCfg.RootConfig.Components {
+			componentsToDelete[i] = component.Name
+		}
+	}
+
+	// Verify if all the components user has specified exist
```

Comment:
I see what you mean here, since we already have checks in place for verifying the components specified in the file so no need to check those, check only those that are user provided.